src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -93,7 +94,8 @@ public void ConnectHeaders_NoNewHeadersToConnect_ShouldReturnNothingToDownload()
 
             ConnectedHeaders connectedHeaders = chainedHeaderTree.ConnectNewHeaders(1, listOfExistingHeaders);
 
-            Assert.True(testContext.ConnectedHeadersIsEmpty(connectedHeaders));
+            Assert.True(testContext.NoDownloadRequested(connectedHeaders));
+            Assert.Equal(11, chainedHeaderTree.GetChainedHeadersByHash().Count);
```

Comment:
The new condition is not really checking that the tree was not modified, but is better than nothing I guess.

Anyway, the test is called  ConnectHeaders_NoNewHeadersToConnect_ShouldReturnNothingToDownload, so if we want to hard is the one that check that the to was not modified, it should be renamed to  ConnectHeaders_NoNewHeadersToConnect_ShouldReturnNothingToDownload_and_NoNewHeaderWasCreatedInTree.